R/translator.R

```diff
@@ -0,0 +1,143 @@
+#' Translator options
+.translator_options <- list(
+  cultural_bignumer_mark = NULL,
+  cultural_punctuation_mark = NULL,
+  cultural_date_format = NULL
+)
+
+#' Translator class
+#'
+#' @field languages character.
+#' @field key_translation character.
+#' @field translation_file character.
+#' @field options list.
+#' @field translations data.frame.
+#' @field translation_language character.
+#'
+#' @return
+#' @export
+#'
+#' @import jsonlite
+#'
+#' @examples
+#' @rdname translator
+Translator <- setRefClass(
+  "Translator",
+  fields = list(
+    languages = "character",
+    key_translation = "character",
+    translation_language = "character",
+    translation_file = "character",
+    options = "list",
+    translations = "data.frame",
+    mode = "character"
+  )
+)
+
+#' @rdname translator
+Translator$methods(
+  initialize = function(translation_file, key_translation) {
+    translation_file <<- translation_file
+    key_translation <<- key_translation
+    options <<- .translator_options
+    # determine the translations format
```

Comment:
This comment is a marker of a section in code. Typically this kind of comments can be avoided and are a sign that there is a function to be extracted